src/pipeline/broad_phase/broad_phase.rs

```diff
@@ -24,6 +24,34 @@ impl ProxyHandle {
     }
 }
 
+/// Proximity handling for BroadPhase updates.
+pub trait BroadPhaseProximityHandler<T> {
+    /// A pre-filter that may cheaply discard objects before checking for bounding volume
+    /// interference.
+    fn allow(&mut self, &T, &T) -> bool;
+
+    /// Actually handle the interference.
+    /// The boolean specifies whether the objects start or stop interacting.
+    fn handle(&mut self, &T, &T, bool);
+}
+
+struct SimpleBroadPhaseProximityHandler<'a, T: 'a> {
+    allow_proximity: &'a mut FnMut(&T, &T) -> bool,
+    handle_proximity: &'a mut FnMut(&T, &T, bool),
+}
+
+impl<'a, T> BroadPhaseProximityHandler<T> for SimpleBroadPhaseProximityHandler<'a, T> {
+    #[inline]
+    fn allow(&mut self, a: &T, b: &T) -> bool {
+        (self.allow_proximity)(a, b)
+    }
+
+    #[inline]
+    fn handle(&mut self, a: &T, b: &T, started: bool) {
```

Comment:
I agree splitting this into two methods `.interference_started(...)` and `.interference_stopped(...)` would be better now that we have a trait-object.